request/private/http-location.rkt

```diff
@@ -7,18 +7,51 @@
 
 (provide
  (contract-out
-  [make-domain-requester (-> string? requester? requester?)]
-  [make-host+port-requester (-> string? exact-nonnegative-integer? requester? requester?)]))
+  [make-domain-requester
+   (-> string? requester? requester?)]
+  [make-host+port-requester
+   (-> string? exact-nonnegative-integer? requester? requester?)]
+  [make-https-requester
+   (-> string? requester? requester?)]))
 
 
-(define (domain+relative-path->http-url domain relative-path)
-  (string->url (format "http://~a/~a" domain relative-path)))
+(define (domain+relative-path->http-url protocol domain relative-path)
+  (string->url (format "~a://~a/~a" protocol domain relative-path)))
 
 (define (host+port->domain host port)
   (format "~a:~a" host port))
 
 (define (make-domain-requester domain requester)
-  (wrap-requester-location (domain+relative-path->http-url domain _) requester))
+  (wrap-requester-location
+   (domain+relative-path->http-url "http" domain _) requester))
+
+(define (make-https-requester domain requester)
```

Comment:
I think `make-https-requester` shouldn't work with a domain and should *only* change the URL of requests made with the underlying requester to HTTPS. Something like `(wrap-requester-location http-url->https-url requester)`.